lib/commands/new.js

```diff
@@ -57,25 +57,21 @@ module.exports = Command.extend({
       commandOptions.skipGit = true;
     }
 
-    if (packageName === '.') {
+    if (projectName === '.') {
       let blueprintName = commandOptions.blueprint === 'app' ? 'application' : commandOptions.blueprint;
       message = `Trying to generate an ${blueprintName} structure in this directory? Use \`ember init\` instead.`;
 
       return Promise.reject(new SilentError(message));
     }
 
-    if (!validProjectName(packageName)) {
-      message = `We currently do not support a name of \`${packageName}\`.`;
+    if (!isValidProjectName(projectName)) {
+      message = `We currently do not support a name of \`${projectName}\`.`;
 
       return Promise.reject(new SilentError(message));
     }
 
     commandOptions.blueprint = normalizeBlueprint(commandOptions.blueprint);
 
-    if (!commandOptions.directory) {
-      commandOptions.directory = packageName;
-    }
```

Comment:
I removed this defaulting so that we can ensure that we _always_ use `--directory` if specified (the defaulting here makes the logic quite a bit harder inside `CreateAndStepIntoDirectory`).